components/validate.js

```diff
@@ -0,0 +1,72 @@
+class Validator {
+    validateSet(set) {
+        var durationCheck = false;
+        var notesCheck = true;
+        var repetitionCheck = false;
+        var weightCheck = true;
+        if (set) {
+            if (set.duration > 0) {
+                durationCheck == true;
+            }
+            if (set.notes.length > 100) {
+                notesCheck == false;
+            }
+            if (set.repetitions > 0) {
+                repetitionCheck == true;
+            }
+            if (!Number.isInteger(set.weight)) {
+                weightCheck == false;
+            }
+        }
+
+        if (durationCheck && notesCheck && repetitionCheck && weightCheck) {
```

Comment:
This could be done in one line `return (durationCheck && notesCheck && repetitionCheck && weightCheck)`